clash-lib/src/Clash/Normalize/Transformations/Letrec.hs

```diff
@@ -439,9 +440,16 @@ topLet (TransformContext is0 ctx) e@(Letrec binds body)
       then return e
       else do
         tcm <- Lens.view tcCache
-        let is2 = extendInScopeSetList is0 (map fst binds)
+        let is2 = extendInScopeSetList is0 (fmap fst binds)
         argId <- mkTmBinderFor is2 tcm (mkUnsafeSystemName "result" 0) body
-        changed (Letrec (binds ++ [(argId,body)]) (Var argId))
+
+        -- TODO Morally this should become
+        --
+        --   Let binds (Let (NonRec argId body) (Var argId))
+        --
+        -- but this causes some tests to fail where they hit more brittle
```

Comment:
Could you mention which tests fail?

Reply:
Turns out no tests anymore. I suspect the ones that did fail were hitting the (previously) broken alpha-equivalence

Reply:
... unless you're running on GHC 8